dashboard/client/src/pages/job/JobDetail.tsx

```diff
@@ -116,17 +116,22 @@ export const JobDetailChartsPage = () => {
     setActorListFilter(undefined);
   };
 
+  // Only display details for Ray Datasets that belong to this job.
+  const jobDatasets = data?.datasets.filter(
+    (dataset) => dataset.job_id === job.job_id,
```

Comment:
can we do the filtering in data_head.py instead? Have the API accept a job_id parameter.

That way the amount of data transferred over the network is reduced